wrappers/services.go

```diff
@@ -252,6 +239,34 @@ func stopService(sysd systemd.Systemd, app *snap.AppInfo, inter Interacter) erro
 	return nil
 }
 
+func serviceIsActivated(app *snap.AppInfo) bool {
+	return len(app.Sockets) > 0 || app.Timer != nil || len(app.ActivatesOn) > 0
+}
+
+func serviceIsSlotActivated(app *snap.AppInfo) bool {
+	return len(app.ActivatesOn) > 0
+}
+
+func serviceUnitsIncludingActivations(app *snap.AppInfo) []string {
```

Comment:
the name is slightly confusing given that this addes the service itself too? anyway it should probably have a comment about what it returns and in which order

Reply:
Good point, I've added docs for this function, if any suggestions for a better name come up, I'm happy to change it

Reply:
I wonder it it would be clearer via this signature?
```go
func serviceUnits(app *snap.AppInfo) (service string, activators []string)
```
then the ordering and implicit assumptions would be explicit and the code below that checks for "len(serviceUnits) == 1" would become "len(activators) == 0"

Reply:
I like your suggestion, I added this